### PR TITLE
Revamping components Along with Fixes

### PR DESCRIPTION
**Pull request by progammer-rkt**
- It revamps components. Components are now categorized into directories where it holds all related hooks, contexts, utilities etc
- Issue 28 - Broken layout
- Issue 27 - When customer login, shipping method disappears
- Issue 26 - All Shipping and Billing Addr API calls reference the cartId from config.js
- Issue 24 - Need Success/Error Message Management
- Issue 21 - Resolve Inconsistencies in React States While Customer Login

### CODE REVIEW

File: src/reactapp/src/components/address/hooks/useCountryState.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { prepareCountryOptions, prepareCountryStateOptions } from '../utility';
+
+export default function useCountryState({ fields }) {
+  const { values, setFieldValue } = useFormikContext();
+  const [{ countryList, stateList }] = useAppContext();
+  const countrySelected = _get(values, fields.country);
+  const regionField = fields.region;
+  const stateSelected = _get(values, regionField);
+
+  // whenever the country is swiched, we need to clear the state input
```
Comment (wigman) [inline src/reactapp/src/components/address/hooks/useCountryState.js:15]: File 52/170 and the first thing I find is a typo in the comment, sorry 😄 

`swiched` => `switched`

File: src/reactapp/src/components/address/hooks/useCountryState.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  useEffect(() => {
+    if (countrySelected) {
+      const stateListContainsStateSelected =
+        !stateList.length ||
+        _get(stateList, countrySelected, []).find(
+          s => s.code === stateSelected
```
Comment (wigman) [inline src/reactapp/src/components/address/hooks/useCountryState.js:21]: As usual, I'm here to say I don't see added value in shortening the variable `state` to `s`. Webpack will rewrite it to a single character anyway, so it costs us nothing to make it more readable with a full variable name 😄

File: src/reactapp/src/components/address/utility/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import _get from 'lodash.get';
+
+import { _objToArray } from '../../../utils';
+
+function sortByItemLabel(item1, item2) {
+  return item1.label.localeCompare(item2.label);
```
Comment (wigman) [inline src/reactapp/src/components/address/utility/index.js:6]: I learned something today 👍

File: src/reactapp/src/components/login/components/LoginFormManager.jsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      }
+
+      setPageLoader(false);
+    } catch (error) {
+      setPageLoader(false);
+      console.log('GuestEmailFormManager', { error });
+    }
+  };
+
+  // Whenever cart-data email info get udpated, the email field will be filled with that value
```
Comment (wigman) [inline src/reactapp/src/components/login/components/LoginFormManager.jsx:148]: typo in `udpated`

**Review by wigman — APPROVED** [review]
Code wise it looks really good. Can't fully judge it functionally until digging into it locally and trying some things out which will hopefully be when I get back from my holiday.
But it looks like a solid job Rajeev!

**Comment by progammer-rkt** [comment]
We will be doing some round of testing before you come back, hopefully. I have helping hands now :)

File: src/reactapp/src/components/address/hooks/useCountryState.js (hunk 1 above)
Comment (progammer-rkt) [reply src/reactapp/src/components/address/hooks/useCountryState.js]: done 👍

File: src/reactapp/src/components/address/hooks/useCountryState.js (hunk 2 above)
Comment (progammer-rkt) [reply src/reactapp/src/components/address/hooks/useCountryState.js]: done 👍

File: src/reactapp/src/components/address/hooks/useCountryState.js (hunk 1 above)
Comment (progammer-rkt) [reply src/reactapp/src/components/address/hooks/useCountryState.js]: done 👍

File: src/reactapp/src/components/login/components/LoginFormManager.jsx (hunk 4 above)
Comment (progammer-rkt) [reply src/reactapp/src/components/login/components/LoginFormManager.jsx]: done 👍

**Pull request merged**